### PR TITLE
boot: Small config entry add changes

### PR DESCRIPTION
**Pull request by medhefgo**
- boot: Remove trivial config entry add helpers
- boot: Rename functions adding entries to match userspace

### CODE REVIEW

File: src/boot/efi/boot.c [hunk 1]
```diff
@@ -2266,18 +2212,22 @@ static void config_entry_add_linux(
...
-                                /* title= */ good_name,
-                                /* loader= */ path,
-                                /* sort_key= */ good_sort_key,
-                                good_version);
+                ConfigEntry *entry = xnew(ConfigEntry, 1);
+                *entry = (ConfigEntry) {
+                        .id = xstrdup(f->FileName),
+                        .type = LOADER_UNIFIED_LINUX,
+                        .title = xstrdup(good_name),
+                        .version = xstrdup(good_version),
```
Comment (keszybz) [inline src/boot/efi/boot.c]: Hmm, are we certain that `good_name` or `good_version` are set? Looking at ` bootspec_pick_name_version_sort_key()`, this doesn't seem guaranteed.

Comment (medhefgo) [reply src/boot/efi/boot.c]: They should be: https://github.com/systemd/systemd/blob/5f1077af1eda0bfee7a7ec68b4912ed8e2d86484/src/fundamental/bootspec-fundamental.c#L49

Comment (medhefgo) [reply src/boot/efi/boot.c]: I guess it's now only guaranteed for good_name. Updated the commit.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**